### PR TITLE
Add Sklearn experiment changes

### PR DESCRIPTION
**Pull request by RafayAK**
This is the code base for a configurable and modular experiment workbench.

Commit features are:

- [x]  Selected datasets from sklearn
- [x]  Select the size of base-dataset
- [x]  Amount of augmented data to add (Gaussian noise with oracle function)
- [x]  Number epochs iterations
- [x]  Define feed-forward network from config
- [x]  WandB integration.
- [x]  Output accuracy and also F1 score, log to wandb
- [x]  Plot distribution of weights and biases.
- [x]  plot decision boundary for a given trained model and log to wandb
- [x]  convert project to pipenv
- [x] Add the ability to log Sklearn dataset plots
- [x]  add examples
- [x]  Create Readme

### CODE REVIEW

**Comment by RafayAK** [comment]
Initial PR ready for merge

**Comment by RafayAK** [comment]
pushed a fix to make sure that data augmentation only applies to the training set.

File: src/data_augmentations/gaussian.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from typing import Tuple
+import numpy as np
+
+
+def augment_data(X: np.ndarray, Y: np.ndarray, noise: float = 0.05, iterations: int = 1) -> Tuple[np.ndarray, np.ndarray]:
```
Comment (alexhernandezgarcia) [inline src/data_augmentations/gaussian.py:5]: I would rename `iterations` as `n_augmentations` or something like that, more explicit.

File: src/data_augmentations/gaussian.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    # instantiate a random number generator.
+    # seed value can be set in the experiment configs globally
+    rng = np.random.default_rng()
+
+    original_X, original_Y = X.copy(), Y.copy()
+
+    new_X, new_Y = X.copy(), Y.copy()
+    for _ in range(iterations):
+        augmented_X, augmented_Y = original_X.copy(), original_Y.copy()
```
Comment (alexhernandezgarcia) [inline src/data_augmentations/gaussian.py:32]: Why not copying directly `X` and `Y` (and omitting the previous copy into `original_` in line 28)

File: src/data_augmentations/gaussian.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        augmented_X += rng.normal(scale=noise, size=augmented_X.shape)
+        new_X, new_Y = np.vstack((new_X, augmented_X)), np.vstack((new_Y, augmented_Y))
+
+    return new_X, new_Y
```
Comment (alexhernandezgarcia) [inline src/data_augmentations/gaussian.py:36]: add line break?

**Review by alexhernandezgarcia — APPROVED** [review]
I have not reviewed and tested every single detail, but in general it looks good to me and I believe we should merge and continue developing from this. We can always make changes and adapt things. Great work, thanks!

File: src/data_augmentations/gaussian.py (hunk 3 above)
Comment (RafayAK) [reply src/data_augmentations/gaussian.py:36]: fixed in #7

File: src/data_augmentations/gaussian.py (hunk 2 above)
Comment (RafayAK) [reply src/data_augmentations/gaussian.py:32]: fixed in #7

File: src/data_augmentations/gaussian.py (hunk 1 above)
Comment (RafayAK) [reply src/data_augmentations/gaussian.py:5]: fixed in #7
